src/controllers/validate.controller.ts

```diff
@@ -0,0 +1,69 @@
+import { NextFunction, Request, Response, Router } from 'express';
+
+import { Controller } from '../interfaces';
+
+import { ParserService } from '../services/parser.service';
+
+import { ProblemException } from '../exceptions/problem.exception';
+import { prepareParserConfig, tryConvertToProblemException } from '../utils/parser';
+
+const BAD_REQUEST = 'Bad Request';
+
+/**
+ * Controller which exposes the Parser functionality, to validate the AsyncAPI document.
+ */
+export class ValidateController implements Controller {
+  public basepath = '/validate';
+
+  private parserService = new ParserService();
+
+  private async validate(req: Request, res: Response, next: NextFunction) {
+    try {
+      const contentType = req.headers['content-type'];
+
+      if (!contentType || (contentType.indexOf('application/json') !== 0 && contentType.indexOf('application/x-yaml') !== 0)) {
+        return next(new ProblemException({
+          type: 'invalid-document-type',
+          title: BAD_REQUEST,
+          status: 400,
+          detail: 'The supported content types are: application/json, application/x-yaml',
+        }));
+      }
+
+      const asyncapi = contentType.indexOf('application/json') >= 0 ? req.body.asyncapi : req.body;
+      if (asyncapi === null) {
```

Comment:
See my previous comment. Also in JS missed/undefined field hasn't `null` value but `undefined`, so you should check in condition `(!asyncapi)` to handle situation with `null`, `undefined` and also with empty string :)

Reply:
yup I was thinking about this too, the `document-validation.middleware` only checked for undefined, but I'll change to !asyncapi 👍